### DISCUSSION TITLE
Unable to run with 32-bit draw indices

### DISCUSSION THREAD

**Discussion by paolotherocker** [ocornut/imgui#5220] [New Users Build/Link/Run issues ONLY!]
I am building ImGui at tip of `docking` branch, with SDL and OpenGL3 backends, and it runs perfectly.

I am using the [ImPlot](https://github.com/epezent/implot) at tip of `master` branch for realtime plots. The ImPlot README strongly recommends using 32-bit indexing, which I understand is enabled by uncommenting the following line from `imconfig.h`
```cpp
#define ImDrawIdx unsigned int
```
I can build fine, but I hit an assertion when calling `IMGUI_CHECKVERSION()`:
```cpp
bool ImGui::DebugCheckVersionAndDataLayout(const char*, size_t, size_t, size_t, size_t, size_t, size_t): Assertion `sz_idx == sizeof(unsigned int) && "Mismatched struct layout!"' failed.
```
I can't see any problem if I don't call `IMGUI_CHECKVERSION()` at all, but I'm assuming that's not recommended?
Apologies if I'm missing something obvious

**Comment by PathogenDavid** [comment]
You definitely should not skip `IMGUI_CHECKVERSION`, ignoring it will become a very confusing problem sooner or later.

Judging by the assert message, it sounds like your application and Dear ImGui are picking up two different versions of `imconfig.h`. Dear ImGui is using `unsigned int` for `ImDrawIdx` but whatever is calling `IMGUI_CHECKVERSION` isn't.

**Reply by paolotherocker** [reply]
@PathogenDavid Thank you very much for your answer, short reply is: you were absolutely right.

I build ImGui as a CMake library for convenience, and to avoid branching out, I had the `ImDrawIdx unsigned int` definition as a compile option in the CMakeLists, instead of modifying `imconfig.h`. The problem was, the compile option doesn't propagate to the project that imports the library, so after copy/pasting the compile option, everything is good. I'll try to have the library export the compile definition, but otherwise this doesn't bother me very much.
